### DISCUSSION TITLE
Font support (code points, bitmaps, etc)

### DISCUSSION THREAD

**Discussion by hpjansson** [hpjansson/chafa#68] [General]
I'm opening this thread for discussing anything having to do with font support. This is only tangentially related to Chafa, but it affects how its output is rendered in various environments.

Ref. discussion started in #66:

* @clort81 mentioned improving PETSCII and other block/legacy symbol support in various publicly available fonts.
* @hpjansson mentioned packaging bitmap fonts as OTB, Terminus specifically.

Feel free to post questions, progress reports, etc. or otherwise coordinate font-related activity here!

**Comment by hpjansson** [comment]
@clort81 If font maintainers are unresponsive, we can try to get your improvements into distro packages, e.g. openSUSE where I'm already a packager.

**Comment by clort81** [comment]
Acknowledged.  Are distros generally dropping .pcf support?  I need pixel-perfect fonts, not smoothed, for terminals displaying small text.

**Reply by hpjansson** [reply]
I think the PCF fonts themselves will still be around for a long time, probably as long as X11. But support is already poor, and it's not going to get better (otoh, doubt it can get much worse). It looks like bitmap-only OpenType is the way forward, though. I did some testing with it yesterday, and it works out of the box on openSUSE. No smoothing as long as you pick the right font size.

I tested with the OTB from here: https://github.com/wyatt8740/IBM3161-font/tree/master/X11 -- just ran gnome-font-viewer on the file and hit "install", then selected it the preferences dialog of xfce-terminal.

**Reply by hpjansson** [reply]
This script worked for me: https://gist.github.com/Earnestly/6bc5bad7666f7bf8816d054b7b76112e

Went from `ter-x*[nb].pcf` -> `Terminus(Bold|Medium).otb`. The result looks exactly right in VTE and works in modern desktop apps. I'll see if I can package this for openSUSE, which doesn't seem to already have it. Maybe other bitmap fonts too.
